### PR TITLE
Directmessaging

### PR DESCRIPTION
**Pull request by MV7757**
Modified Message.Java in order to create a recipient field and accept it as a parameter in the constructor. Also, created a getter method for the Message class. Since I changed the constructor, I modified where the Message class was called. This ended up being MessageServlet.java and DataStore.java In message servlet I added the redirect to include recipient and created a recipient variable through a request. In datastore, I changed the constructor to include the new recipient variable.

### CODE REVIEW

**Comment by nerdatmath** [comment]
> Steps 1-4 on Direct messaging

The description of the pull request should explain the intent of the change. What is the change intended to achieve? With the current description, the reader has to search to find what you mean by "Direct Messaging", and what steps 1-4 are.

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public void storeMessage(Message message) {
     messageEntity.setProperty("user", message.getUser());
     messageEntity.setProperty("text", message.getText());
     messageEntity.setProperty("timestamp", message.getTimestamp());
-
```
Comment (nerdatmath) [inline src/main/java/com/google/codeu/data/Datastore.java:45]: You're dropping the empty line between the setProperty calls and the call to datastore.put. Is that intended?

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 2]
```diff
@@ -33,7 +33,6 @@
 /** Handles fetching and saving {@link Message} instances. */
 @WebServlet("/messages")
 public class MessageServlet extends HttpServlet {
-
```
Comment (nerdatmath) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java:36]: Did you intend to remove this blank line?

File: src/main/webapp/index.html [hunk 3]
```diff
@@ -34,5 +34,8 @@ <h1>CodeU Starter Project</h1>
     their URL.</p>
     <p>This is your code now! Feel free to make changes, and don't be afraid to get creative!
     You could start by modifying this page to tell the world more about your team.</p>
+
```
Comment (nerdatmath) [inline src/main/webapp/index.html:37]: I believe this is an unrelated change; please make it in a different pull request.

**Review by nerdatmath — DISMISSED** [review]

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 1 above)
Comment (MV7757) [reply src/main/java/com/google/codeu/data/Datastore.java:45]: This was not something that was intended, do I add back in the new line?

File: src/main/java/com/google/codeu/servlets/MessageServlet.java (hunk 2 above)
Comment (MV7757) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java:36]: Thats typically how I do it, but if I should keep it then I can change it!

File: src/main/webapp/index.html (hunk 3 above)
Comment (MV7757) [reply src/main/webapp/index.html:37]: How do I take this out of this request, and put it in a new one?

**Review by michzhang — APPROVED** [review]
Everything is resolved, the provided code isn't something we can edit. Looks good!
